compiler/qsc_rca/src/rca.rs

```diff
@@ -0,0 +1,2340 @@
+                expr_id,
+                applications_table,
+                package,
+                package_scaffolding,
+            )
+        }
+        StmtKind::Item(_) => {}
+    }
+    package_scaffolding
+        .stmts
+        .insert(stmt_id, applications_table.clone())
+}
+
+fn simulate_block(
+    id: StoreBlockId,
+    application_instance: &mut ApplicationInstance,
+    package_store: &PackageStore,
+    package_store_scaffolding: &mut PackageStoreScaffolding,
+) {
+    // This function is only called when a block has not already been analyzed.
+    if package_store_scaffolding.find_block(id).is_some() {
+        panic!("block is already analyzed");
+    }
+    if application_instance.blocks.get(&id.block).is_some() {
+        panic!("block is already analyzed in application instance");
+    }
+
+    // Initialize the block's compute kind.
+    let block = package_store.get_block(id);
+    let mut block_compute_kind = ComputeKind::Classical;
+
+    // Iterate through the block statements and aggregate the runtime features of each into the block's compute kind.
+    for stmt_id in block.stmts.iter() {
+        let store_stmt_id = StoreStmtId::from((id.package, *stmt_id));
+        simulate_stmt(
+            store_stmt_id,
+            application_instance,
+            package_store,
+            package_store_scaffolding,
+        );
+        let stmt_compute_kind = application_instance
+            .stmts
+            .get(stmt_id)
+            .expect("statement's compute kind should exist");
+        block_compute_kind =
+            aggregate_compute_kind_runtime_features(block_compute_kind, stmt_compute_kind);
+    }
+
+    // Update the block's value kind if its non-unit.
+    if block.ty != Ty::UNIT {
+        let last_stmt_id = block
+            .stmts
+            .last()
+            .expect("block should have at least one statement");
+        let last_stmt_compute_kind = application_instance
+            .stmts
+            .get(last_stmt_id)
+            .expect("statement's compute kind should exist");
+
+        // We only have to update the block's value kind if the last statement's compute kind is quantum and dynamic.
+        if let ComputeKind::Quantum(last_stmt_quantum_properties) = last_stmt_compute_kind {
+            if let ValueKind::Dynamic = last_stmt_quantum_properties.value_kind {
+                // If the block's last statement's compute kind is quantum, the block's compute kind must be quantum too.
+                let ComputeKind::Quantum(block_quantum_properties) = &mut block_compute_kind else {
+                    panic!("block's compute kind should be quantum");
+                };
+
+                // The block's value kind must be static since this is the first time a value kind is set.
+                let ValueKind::Static = block_quantum_properties.value_kind else {
+                    panic!("block's value kind should be static");
+                };
```

Comment:
Since there is no assignment being used from the let-binding, this can just be an assert.
```suggestion
                assert!(block_quantum_properties.value_kind == ValueKind::Static, "block's value kind should be static");
```